ocaml/xapi/xapi_vm.ml

```diff
@@ -376,6 +392,8 @@ let start ~__context ~vm ~start_paused ~force =
     Pool_features.assert_enabled ~__context ~f:Features.Network_sriov ;
   if not force then
     assert_memory_constraints ~__context ~vm vmr.API.vM_platform ;
+  if not (Pool_features.is_enabled ~__context Features.DMC) then
```

Comment:
This is too late, because the memory checks (to see if the VM fits on the host) have already been done at this point. This happens in `Xapi_vm_helpers.assert_can_boot_here` in the `assert_enough_memory_available` call, and this check is based on dynamic_min. It is called through `forward_to_suitable_host` (for `VM.start`) or `reserve_memory_for_vm` (for `VM.start_on`) in message_forwarding. Those functions are also used for operations like resume and migrate, so we can't call `enforce_memory_constraints` inside of them. The safest place for it is probably in `VM.start` and `VM.start_on` in message_forwarding.